08/kth-missing-positive-number.md

```diff
@@ -179,35 +294,98 @@ def test_empty_lists():
 * topics: pse
 ##### !question
 
-Without writing code, describe how you would implement `kth_missing_positive_number` in enough detail that someone else could write the code. 
+Without writing code, describe how you would implement `find_missing_positive_number` in enough detail that another developer could reasonably implement a solution. We should capture the main use cases, but the steps do not need to be a detailed plan for every contingency. 
+- The objective is to create a roadmap that we can use to keep ourselves oriented towards our goal
+- It is okay to leave some of the finer details to be worked out in the implementation itself!
 
-* It may be helpful to break up the problem/algorithm into smaller subproblems/algorithms. For example, 1. Handle invalid input, 2. Given valid input, perform the computation/solve the problem/etc.
-* Your logical steps could take the form of a numbered list, pseudo code, or anywhere in between. What's important at this stage is to think through and outline the implementation before writing code.
+As you write your steps, keep the following guidelines in mind:
+* We want to think about a general approach rather than what the code would look like line-by-line. 
+* It may be helpful to break up the problem/algorithm into smaller subproblems/algorithms. 
+    * For example: 1. Handle edge cases, 2. Perform the computation/solve the problem/etc.
+* The steps should be a description as if you were talking out the problem with another person and should be agnostic of any particular language. 
+    * As such, they should not include code syntax in the description.
 
-##### !end-question
+What's important at this stage is to think through and outline the implementation before writing code.
 
+##### !end-question
 ##### !placeholder
 
 Write the logical steps here.
 
 ##### !end-placeholder
-
 ### !explanation
 
 Example Steps for an O(n) solution:
 
-1. Check for valid input
-2. If `k` is smaller than the first value in the input list `arr`, return `k`
-3. Create a loop over the input `arr` that: 
-   * starts at the first element of `arr` 
-   * ends one element before the last element of `arr`
-4. Inside the loop:
-    1. Check if there are numbers missing between the current element and the next element in `arr`. Store the difference between the next element and the current element of `arr` in a new variable `missing_count`.
-    2. if `k` is less than or equal to `missing_count`, return the value of `arr` at `index` plus `k` 
-    3. Update the variable `k` by subtracting `missing_count`
-5. If the loop ends, return the last element of the list plus the remaining value of `k`.
+1. Handle edge case:
+    * If `kth_missing` is smaller than the first value in `numbers`, then `kth_missing` is itself the missing number. Return `kth_missing`.
```

Comment:
Edge case for empty list, which out current linear time implementation would fail for.
```suggestion
    * If `numbers` is empty, then `kth_missing` is itself the missing number. Return `kth_missing`.
        * Since every number is missing, and the numbers and `kth_missing` both start from `1`, `kth_missing` is the missing number.
    * If `kth_missing` is smaller than the first value in `numbers`, then `kth_missing` is itself the missing number. Return `kth_missing`.
```